general/development/policies/deprecation/index.md

```diff
@@ -135,16 +137,18 @@ throw new coding_exception(
 
 ```php
 #[\core\attribute\deprecated('foobar::blah()', since: '4.4', mdl: 'MDL-XXXXX', final: true)]
-public function foobar(): void {
+public function foobar(int $old, array $params): array {
     \core\deprecation::emit_deprecation_if_present([self::class, __FUNCTION__]);
+    return [];
 }
 ```
 
 </TabItem>
 
 </Tabs>
 
-- All function parameters should be removed.
+- When deprecating methods in a class, retain the existing parameters and return type.
```

Comment:
Honestly I’d just simplify it further. Make it the same for class methods, and functions. There is no benefit in changing the signature so just leave the whole thing

Reply:
Okay, will change it over.